csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -48,21 +48,28 @@ public BinaryArray(ArrowTypeId typeId, ArrayData data)
             data.EnsureBufferCount(3);
         }
 
-        public abstract class BuilderBase<TArray, TBuilder> : IArrowArrayBuilder<byte, TArray, TBuilder>
+        public abstract class BuilderBase<T, TArray, TBuilder> : IArrowArrayBuilder<T, TArray, TBuilder>
+            where T: struct
             where TArray : IArrowArray
-            where TBuilder : class, IArrowArrayBuilder<byte, TArray, TBuilder>
+            where TBuilder : class, IArrowArrayBuilder<T, TArray, TBuilder>
         {
+
             protected IArrowType DataType { get; }
             protected TBuilder Instance => this as TBuilder;
             protected ArrowBuffer.Builder<int> ValueOffsets { get; }
-            protected ArrowBuffer.Builder<byte> ValueBuffer { get; }
+            protected ArrowBuffer.Builder<T> ValueBuffer { get; }
+            protected BooleanArray.Builder ValidityBuffer { get; }
+
             protected int Offset { get; set; }
+            protected int ValidityOffset { get; set; }
```

Comment:
`ValidityOffset` doesn't appear to be used.

Reply:
Fixed